client/packages/programs/src/JsonForms/common/components/Array.tsx

```diff
@@ -291,104 +463,94 @@ const ArrayComponent = (props: ArrayControlCustomProps) => {
+                  <ConfirmationModal
+                    open={removeIndex !== undefined}
+                    onConfirm={() => {
+                      if (removeIndex !== undefined)
+                        handleRemoveItem(removeIndex);
+                    }}
+                    onCancel={() => setRemoveIndex(undefined)}
+                    title={t('label.remove')}
+                    message={t('messages.confirm-remove-item')}
+                  />
+
+                  <IconButton
+                    icon={isEditable ? <MinusCircleIcon /> : null}
+                    label={t('label.remove')}
+                    color="primary"
+                    className="array-remove-icon"
+                    sx={{ visibility: 'hidden' }}
+                    onClick={e => {
+                      setRemoveIndex(index);
+                      // Don't toggle the accordion:
+                      e.stopPropagation();
+                    }}
+                  />
+                  {itemLabelComponent}
+                </Box>
+              </AccordionSummary>
+              <AccordionDetails>
+                <JsonFormsDispatch
+                  key={childPath}
+                  schema={schema}
+                  uischema={childUiSchema || uischema}
+                  enabled={isEditable}
+                  path={childPath}
+                  renderers={renderers}
                 />
-                <Typography
-                  sx={{
-                    fontWeight: 'bold',
-                    textAlign: 'end',
-                    whiteSpace: 'nowrap',
-                  }}
-                >
-                  {uischema?.itemLabel
-                    ? RegexUtils.formatTemplateString(
-                        uischema?.itemLabel,
-                        {
-                          ...(typeof child === 'object' ? child : {}),
-                          index: index + 1,
-                        },
-                        ''
-                      )
-                    : index + 1}
-                </Typography>
-              </Box>
-            </AccordionSummary>
-            <AccordionDetails>
-              <JsonFormsDispatch
-                key={childPath}
-                schema={schema}
-                uischema={childUiSchema || uischema}
-                enabled={enabled}
-                path={childPath}
-                renderers={renderers}
-              />
-            </AccordionDetails>
-          </Accordion>
-        );
-      })}
+              </AccordionDetails>
+            </Accordion>
+          );
+        })
+        .sort((a, b) =>
+          uischema.options?.['reverse']
+            ? (b.key as number) - (a.key as number)
+            : (a.key as number) - (b.key as number)
+        )}
     </Box>
   );
 };
 
-export const Array = withJsonFormsArrayControlProps(
+export const ArrayControl = withJsonFormsArrayControlProps(
   ArrayComponent as ComponentType<ArrayControlProps>
```

Comment:
Do you remember why this `as` cast was needed?

Reply:
I *think* it's because the `withJsonFormsArrayControlProps` method is expecting the standard ArrayControlProps, but our versions has a few extra options etc.